### PR TITLE
Update Monitor Renames and Enums

### PR DESCRIPTION
**Pull request by nisha-bhatia**
This PR addresses the following: 
- [ ] Add endpoint as a public property
- [ ] Change filter parameter name in CreateQuery to query
- [x] Start ServiceVersion numbering at 1 in LogsQueryClientOptions
- [x] Start ServiceVersion numbering at 1 in MetricsQueryClientOptions
- [x] Change Top to Size in MetricsQueryOptions
- [ ] Add partial failure plan to LogsQueryResult
- [ ] GetResult in class LogsBatchQueryResult should return a LogsBatchQueryResult
- [ ] Decide if GetResult in class LogsBatchQueryResult is necessary
- [x] Hide Deserialize<T> in class LogsTable
- [x] Make GetObject methods internal in class LogsTableRow
- [ ] Don't return nullables everywhere
- [ ] Rename NamespaceCollection with Metric prefix
- [x] Spell out Qos in NamespaceClassification
- [ ] Rename QueryModelFactory to MonitorQueryModelFactory - change title in autorest.md
- [ ] Rename TimeSeriesElement with Metric prefix?
- [x] Change Data to Values in TimeSeriesElement class
- [ ] Add Collection type

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
Linked: https://github.com/Azure/azure-sdk-for-net/issues/23580

File: sdk/monitor/Azure.Monitor.Query/src/Generated/Models/NamespaceClassification.cs [hunk 1]
```diff
@@ -30,8 +31,10 @@ public NamespaceClassification(string value)
         public static NamespaceClassification Platform { get; } = new NamespaceClassification(PlatformValue);
         /// <summary> Custom. </summary>
         public static NamespaceClassification Custom { get; } = new NamespaceClassification(CustomValue);
+
         /// <summary> Qos. </summary>
-        public static NamespaceClassification Qos { get; } = new NamespaceClassification(QosValue);
+        [CodeGenMember("Qos")]
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/src/Generated/Models/NamespaceClassification.cs:36]: You would need to define a partial struct outside of generated directory and applying this attribute there. Otherwise running the generator would reset your changes.

File: sdk/monitor/Azure.Monitor.Query/src/Models/ResultType.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@ namespace Azure.Monitor.Query.Models
 {
     internal enum ResultType
     {
-        /// <summary> Data. </summary>
+        /// <summary> Values. </summary>
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/src/Models/ResultType.cs:8]: Seems unrelated.

**Review by pakrym — APPROVED** [review]
